### PR TITLE
Allow users to only translate and review one story at a time per language

### PR DESCRIPTION
**Pull request by jennchenn**
## Notion ticket link

[Allow users to only translate and review one story at a time per language](https://www.notion.so/uwblueprintexecs/Allow-users-to-only-translate-and-review-one-story-at-a-time-per-language-b0b40542f8f44f47bf3490e03004491e)

## Implementation description
_backend changes_:
- Added checks before assigning user as a translator/reviewer to ensure they have not already been assigned to translate/review a story of the same language
- Modified the `get_story_translations` query parameters to all be optional
- Updated `insert_test_data.py` script
_frontend changes_:
- Updated frontend to query for and display all stories user is currently translating/reviewing in homepage "My Work" + disabled filters 
- Added translator/reviewer pill to story card

## Steps to test

1. Rebuild docker containers; clear +re-seed database:
```
$ docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data erase"
$ docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"
```
2. Login to frontend (these steps will use Dwight D. Eisenhower as the example user)
3. Verify that in the "My Work" tab, all the stories the user is translating/reviewing show up:
  - Pride and Prejudice (English UK) [Translator]
  - To Kill a Mockingbird (German) [Translator]
  - The Great Gatsby (English US) [Reviewer]
image
4. Check that you are able to click into any of the stories and view the translation/review platform without issues
5. Click on browse stories and try to translate a story the user has level access for, but in a language the user is already translating; this should result in an error `Error: User can't be assigned as a translator`
  - For Eisenhower try English UK <= level 4 or German <= level 3
image
6. Try to translate a story the user has level access for, but in a language the user is *not* currently translating
  - Try a story in English US <= level 4
7. Repeat steps 5 and 6 for reviewing stories
  - Eisenhower can review English UK <= level 3 (this should work without issues) and English US <= level 4 (this should fail as he is already reviewing a story in this language)
8. Navigate back to the "My Work" tab and verify that any stories the user was successfully added as a translator/reviewer appear with the correct role pill 

disabled filter

## What should reviewers focus on?

- Design of the story card role pill/disabled filters
- Checks added to `story_service.py`
- Data seeded in `insert_test_data.py` is consistent (i.e. users seeded in db are only translating and reviewing at most 1 story per language)
- Is `seed_db.sh` still being used by the team? Should I update the data there as well?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/services/implementations/story_service.py [hunk 1]
```diff
@@ -343,3 +365,24 @@ def _get_num_approved_lines(self, translation_contents):
...
+            StoryTranslation.query.filter(StoryTranslation.reviewer_id == user_id)
+            .filter(StoryTranslation.stage != "PUBLISH")
+            .all()
+        )
+
+    def _get_story_translation_languages(self, story_translations):
+        languages = set(
+            [story_translation.language for story_translation in story_translations]
+        )
+        print(languages)
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:387]: nit: remove

File: backend/python/app/services/implementations/story_service.py [hunk 2]
```diff
@@ -97,18 +109,24 @@ def create_translation(self, translation):
             raise error
         return new_story_translation
 
-    def get_story_translations(self, user_id, translator, language, level):
+    def get_story_translations(self, user_id, is_translator, language, level, stage):
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:112]: when in the frontend is `stage` getting passed into this service?

**Review by gaoxk — CHANGES_REQUESTED** [review]
nice! 
1. There appears to be a bug with displaying story cards. only 4 are showing up for dwight, but I have him involved in 5 story_translations. I think it's because `a tale of two cities `(id 3) needs to show up twice
image
image
2. You can faintly see that a language, role, and access level are selected. Can you have nothing selected? (language can just be blank). If when looking into it, you believe it should be it's own pr, lmk and we can make a separate ticket.
image

File: backend/python/app/services/implementations/story_service.py (hunk 2 above)
Comment (jennchenn) [reply backend/python/app/services/implementations/story_service.py:112]: oops it is not being used i'll remove it--i was originally planning to use it for the backend checks but then decided it makes more calls to the db than necessary so made separate helper functions for it

**Comment by jennchenn** [comment]
>     1. There appears to be a bug with displaying story cards. only 4 are showing up for dwight, but I have him involved in 5 story_translations. I think it's because `a tale of two cities `(id 3) needs to show up twice
I changed the way we are joining the results and returning them from the `get_story_translations` query so it should return all the story translations now

>     2. You can faintly see that a language, role, and access level are selected. Can you have nothing selected? (language can just be blank). If when looking into it, you believe it should be it's own pr, lmk and we can make a separate ticket.

I was able to get things to be deselected for the Select component by making an empty string option, and for the buttons by passing in an `isDisabled` prop and changing the style. I think a better way would be using the Chakra UI props, but I'm not sure what that'd look like (I think Checkbox has an `isChecked` prop but I wasn't able to figure out how to set it manually)
image

**Review by gaoxk — APPROVED** [review]
nice! the isDisabled pattern looks good to me - buttons and such have a built-in disabled styling, but your pattern lets us keep our disabled styling consistent.

**Pull request merged**